modules/ROOT/pages/policies-compare-mule3-and-mule4.adoc

```diff
@@ -0,0 +1,40 @@
+include::_attributes.adoc[]
+endif::[]
+:keywords: policy, custom, ootb, offline
+
+In Mule versions 3 and 4, the policies remain the same, with a few exceptions. The following section describes the similarities and differences in these Mule versions.
+
+// Policies in Mule 4
+// Classloader isolation exists between the application, the runtime and connectors, and policies. To fulfill the isolation requirement, policies are self-contained and packaged to include libraries needed for execution, similar to an application. Additionally, certain changes to the packaging, size, and scope of a policy are introduced in Mule 4.
+
+[%header,cols="40a,50a,50a"]
+|===
+| Feature | Mule 3 | Mule 4 
+|           | Uses MEL expressions | Uses DataWeave 2.0 
+| Policies distribution | Policies cannot be distributed outside of runtime | Policies can be distributed outside of the runtime, which simplifies upgrades 
+| Policies  | Partial non-blocking processing implemented for policies | * All policies are non-blocking to improve performance
+* Versioning of policies is introduced
+| Policy execution | Throttling, Rate Limiting, and CORS policies executed first. Other policies can be reordered | All policies, except CORS, can be reordered 
+| Resource Level Policy Support | Restricted to RAML-based APIs | Support extended to HTTP-based API, along with RAML-based APIs 
+| Class loader isolation  | No class loader isolation supported | Class loader isolation exists between application libraries, extensions, and runtime libraries 
+| Policy change | Throttling | Added the Spike Control policy, which uses an algorithm based on the sliding window mechanism to focus exclusively on reducing spikes in traffic and protecting a Mule Runtime instance. 
+| Authentication policies | HTTP Basic Authentication policy depended on the Simple Security Manager or LDAP Security Manager policies | Simple Basic Authentication and LDAP Basic Authentication policies include Security Manager configuration
+| Rate Limiting policy | The policy can share the quota between cluster nodes. Quota not configurable. | * Provides the ability to configure an identifier for better flexibility. Capabilities that can be optionally configured include:
+* Ability to return headers (Expose Headers checkbox)
+* Ability to Clusterize the quota 
+| Policy configuration | Header propagation occurs be default  | * Header propagation no longer occurs by default. * Header propagation can be configured for Token Enforcement, Federation, and Rate Limiting policies by checking the Expose Headers checkbox.
+| Custom Policies | Not available in Mule 3| * Powerful engine for building custom policies, using the XML schema language
+* improved error handling
+* Support for all Mulesoft components
+* Isolation between policies and application
+* Can be applied to outbound requests, coming back from the backend services
+* Can package additional libraries
+* Policies can be uploaded to Exchange
```

Comment:
Policies can be it's actually policies are uploaded to exchange and therefore support for versioning is now included.

Reply:
Done!